### PR TITLE
Feature/peramb merge

### PR DESCRIPTION
**Pull request by felixerben**
please do not merge yet. current version of the merge of Perabulator.hpp and PerambFromSolve.hpp

### CODE REVIEW

File: Hadrons/Modules/MDistil/Perambulator.hpp [hunk 1]
```diff
@@ -41,15 +41,21 @@ BEGIN_MODULE_NAMESPACE(MDistil)
  *                             Perambulator                                    *
  ******************************************************************************/
 
+GRID_SERIALIZABLE_ENUM(Mode, undef, perambOnly, 0, inputSolve, 1, outputSolve, 2);
```
Comment (aportelli) [inline Hadrons/Modules/MDistil/Perambulator.hpp:44]: not in top namespace, possibly in module class

File: Hadrons/Modules/MDistil/Perambulator.hpp [hunk 2]
```diff
@@ -89,22 +95,28 @@ TPerambulator<FImpl>::TPerambulator(const std::string name) : Module<Perambulato
 template <typename FImpl>
 std::vector<std::string> TPerambulator<FImpl>::getInput(void)
 {
-    return {par().lapevec, par().solver, par().noise, par().DistilParams};
+    std::vector<std::string> out={par().lapevec, par().solver, par().noise, par().DistilParams};
+    Mode perambMode{par().perambMode};
+    if(perambMode == Mode::inputSolve)
+    {
+        std::cout<< "unsmeared solves are an input" << std::endl;
```
Comment (aportelli) [inline Hadrons/Modules/MDistil/Perambulator.hpp:102]: don't use `std::cout`

File: Hadrons/Modules/MDistil/Perambulator.hpp [hunk 3]
```diff
@@ -182,55 +195,58 @@ void TPerambulator<FImpl>::execute(void)
...
                 for (int ds = 0; ds < dp.SI; ds++)
                 {
-                    LOG(Message) <<  "LapH source vector from noise " << inoise << " and dilution component (d_k,d_t,d_alpha) : (" << dk << ","<< dt << "," << ds << ")" << std::endl;
-                    dist_source = 0;
-                    evec3d = 0;
-                    for (int it = dt; it < Nt; it += dp.TI)
-                    {
-                        const int t_inv{(dp.tsrc + it)%Nt};
-                        if( t_inv >= Ntfirst && t_inv < Ntfirst + Ntlocal )
+                    if(perambMode == Mode::inputSolve){
```
Comment (aportelli) [inline Hadrons/Modules/MDistil/Perambulator.hpp:198]: accolade style

**Comment by felixerben** [comment]
I went through this again and I think it is as ready for develop as it can be for now. We will obviously need some changes as soon as the DistilNoise class is implemented, but that should be straightforward.

**Comment by felixerben** [comment]
The things this pull request improves:
1) PerambFromSolve module is obsolete and part of Perambulator now
2) New module to load Laplacian eigenvectors
3) Overall more readable code
4) Perambulator objects have a new dataset for the noise class, which will be needed for Nelson's meson field module

**Comment by felixerben** [comment]
The test I've done was to compute perambulator objects from random lattices on my laptop, and I get bit-identical results for the old module, and both ways to compute it in the new module (directly or by saving the solve first, then computing the perambulator from the solve in another module)

**Pull request merged**